--- crates/nes-core/src/cheat_codes.rs
+++ crates/nes-core/src/cheat_codes.rs
@@ -342,6 +342,21 @@
         // compare = 0x70 | 0x80 | 0x07 | 0x08 = 0xFF
         assert_eq!(code_n7_n6_n5.compare(), Some(0xFF));
     }
+
+    #[test]
+    fn cheat_code_decodes_value_bits_with_zeroes() {
+        let code_all_zeroes = CheatCode::from_str("AAAAAA").unwrap();
+        assert_eq!(code_all_zeroes.value(), 0);
+        assert_eq!(code_all_zeroes.address(), 0x8000);
+    }
+
+    #[test]
+    fn cheat_code_decodes_value_bits_with_alternating_bits() {
+        // We want to test bitwise ops vs XOR
+        let code_alt = CheatCode::from_str("PZLGITYE").unwrap();
+        assert_eq!(code_alt.address(), 37042);
+        assert_eq!(code_alt.value(), 34);
+        assert_eq!(code_alt.compare(), Some(119));
+    }
 }

 #[cfg(test)]
